adapters/adorsys-integ-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/service/provider/OauthHeaderInterceptor.java

```diff
@@ -20,25 +20,34 @@
 import de.adorsys.xs2a.adapter.service.RequestHeaders;
 import de.adorsys.xs2a.adapter.service.config.AdapterConfig;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
 public class OauthHeaderInterceptor implements Request.Builder.Interceptor {
 
+    private static final String BANK_CODE_FOR_OAUTH = "adorsys.oauth_approach.bank_code";
     static final String OAUTH_HEADER_NAME = "adorsys.oauth_approach.header_name";
     static final String OAUTH_HEADER_VALUE = "adorsys.oauth_approach.header_value";
-    static final String BANK_CODE_FOR_OAUTH = "adorsys.oauth_approach.bank_code";
 
     @Override
     public Request.Builder apply(Request.Builder builder) {
         String oauthBankCode = AdapterConfig.readProperty(BANK_CODE_FOR_OAUTH, "");
+        List<String> bankCodes = Arrays.stream(oauthBankCode.split(","))
+                                     .map(String::trim).collect(Collectors.toList());
         String requestBankCode = builder.headers().get(RequestHeaders.X_GTW_BANK_CODE);
 
-        if (oauthBankCode.isEmpty() || !oauthBankCode.equals(requestBankCode)) {
+        if (oauthBankCode.isEmpty() || !bankCodes.contains(requestBankCode)) {
             return builder;
         }
 
+        int idx = bankCodes.indexOf(requestBankCode);
```

Comment:
This indexing logic is way too complicated. Why not just add another configuration property?

Reply:
because it is a temporary workaround

Reply:
tomorrow they can ask to add one more value and we would add more parameters. So it's a trade-off between complexity and flexibility